libenv/sysinfo.c

```diff
@@ -3335,6 +3335,93 @@ void GetDefVars(EvalContext *ctx)
         CF_DATA_TYPE_STRING, "invocation,source=agent,command_name=jq");
 }
+
+/*****************************************************************************/
+
+static void SysOSNameHuman(EvalContext *ctx)
+{
+    // This function sets the $(sys.os_name_human) variable
+    assert(ctx != NULL);
+    const char *const lval = "os_name_human";
+
+    /**
+     * Order is important. The first if-statement is more specific and wins.
+     * E.g. prefer Ubuntu over Debian.
+     */
+    if (EvalContextClassGet(ctx, NULL, "ubuntu") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "Ubuntu", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=ubuntu");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "debian") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "Debian", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=debian");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "centos") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "CentOS", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=centos");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "fedora") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "Fedora", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=fedora");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "redhat") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "RHEL", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=redhat");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "aix") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "AIX", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=aix");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "hpux") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "HP-UX", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=hpux");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "suse") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "SUSE", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=suse");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "macos") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "macOS", CF_DATA_TYPE_STRING,
```

Comment:
*MacOS*

Reply:
Are you sure? They tend to use lower case 'm' in the OS name. E.g. "macOS Big Sur"

Reply:
@larsewi @vpodzime lowercase is correct

Reply:
*macOS, previously Mac OS X...* :man_facepalming: